### PR TITLE
[red-knot] Understand `typing.Type`

### LINKED ISSUE #14891: [red-knot] Understand `typing.Type`
Subtask of #14172.

### PR DESCRIPTION
**Pull request by InSyncWithFoo**
## Summary

Resolves #14891.

## Test Plan

Markdown tests.

### CODE REVIEW

**Comment by AlexWaygood** [comment]
> This has one small problem: Inheriting from `Type` results in an MRO with both `type` and `Generic`, but `Mro::of_class_impl()` expects only one returned value from `ClassBase::try_from_ty()`. Fixing this would require quite a few significant changes here and there; should I try to do so, or should it be left as a TODO for now?

Fine to just TODO all that for now. We don't even understand that `typing.Generic` is a class at all right now!

**Review by AlexWaygood — APPROVED** [review]
Thanks!

File: crates/red_knot_python_semantic/resources/mdtest/type_of/typing_dot_Type.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## Annotation
+
+`typing.Type` can be used interchangeably with `type`:
+
+```py
+from typing import Type
+
+class A: ...
+
+def _(c: Type, d: Type[A], e: Type[A]):
```
Comment (carljm) [inline crates/red_knot_python_semantic/resources/mdtest/type_of/typing_dot_Type.md:12]: Why does the `e` argument exist here?

**Comment by carljm** [comment]
Nice, thank you!

File: crates/red_knot_python_semantic/resources/mdtest/type_of/typing_dot_Type.md (hunk 1 above)
Comment (AlexWaygood) [reply crates/red_knot_python_semantic/resources/mdtest/type_of/typing_dot_Type.md:12]: Sorry, I think I screwed that up when simplifying the mdtests a little :-)

**Pull request merged**